src/api/app/models/workflow/step.rb

```diff
@@ -15,7 +15,15 @@ def call(_options)
   end
 
   def target_project_name
-    "home:#{@token.user.login}:#{source_project_name}:PR-#{scm_webhook.payload[:pr_number]}"
+    return step_instructions[:target_project] if scm_webhook.push_event?
+
+    pr_subproject_name = if scm_webhook.payload[:scm] == 'github'
```

Comment:
isn't too optimistic to assume that this whole chain of commands will never fail?

Reply:
why? `target_repository_full_name` and `path_with_namespace` are always part of the payload, and there will always be one and only one `/`.

Reply:
> there will always be one and only one `/`.

True :+1:. Slashes are replaced by dashes (`-`) in repository names.